cloudevents/working-drafts/minor-versions.md

```diff
@@ -0,0 +1,337 @@
+that it will create a v1.1 event accidentally after an SDK upgrade - and that
+could cause problems. (The SDK handling of minor versions is relatively
+complex, and discussed later.) But if they produce a v1.0 event with an
+`xyz` attribute with value "test", that's fine: it's still a valid v1.0 event.
+
+An event producer that is aware of v1.1 should ensure that any use of the `xyz`
+attribute is valid - in other words, it can't have values of "test"
+(wrong type) or 123 (constraint violation). That would be an invalid v1.1 event
+in the same way that specifying a `time` attribute value of "lunchtime" would be
+invalid.
+
+## Event proxies
+
+Proxies which are unaware of v1.1 should be able to proxy v1.1 events without
+any issues, *unless* they attach special meaning to unspecified attributes (whether
+`xyz` or anything else). For example, a proxy which makes use of the `sequence` extension
+attribute but isn't aware of v1.0 cannot safely assume that v1.1 events will use `sequence`
+in the same way as v1.0. (`sequence` may become a core attribute with a particular meaning.)
+The proxy could make a choice of:
+
+- Expecting that the previous attribute *is* used in the same way, and treating
+  it as per v1.0 events, with a suitable failure mode.
+- Not attaching any significance to the attribute for v1.1 events, treating it as
+  any other extension attribute.
+
+There's no one-size-fits-all approach here.
+
+## Event consumers
+
+An event consumer that doesn't know about v1.1 should still be able to consume
+a v1.1 event without any knowledge of the `xyz` attribute; it should be treated as a regular
+extension attribute. However, the consumer does "know" that the event is v1.1, so can be aware
+that `xyz` (and any other extension attributes) *may* be a core attribute in v1.1.
+(In other words, they can distinguish between `xyz` appearing in a v1.0 event which is *definitely*
+an extension attribute as the consumer is aware of all core v1.0 attribtes,
+and `xyz` appearing in a v1.1 event which is *either* an extension attribute *or* a new optional
+core attribute.)
+
+## SDKs
+
+If we adopt minor versions, SDKs should change to support them, so that (for example) an SDK
+supporting v1.1 will *also* parse v1.2 events. We should keep track of which SDKs
+(and which versions of those SDKs) support parsing v1.x events.
+
+Different SDKs expose features very differently. This document is written from the perspective
+of the C# SDK, but other viewpoints would be highly valued: please add feedback, particularly
+if the approach suggested doesn't make sense for an SDK you're familiar with.
+
+### Creating events: default version handling
+
+It may sound natural to allow something like:
+
+```csharp
+var evt = new CloudEvent { Id = "..." };
+```
+
+... to just use the latest spec version that the SDK supports. However, this is a breaking change.
+Consider:
+
+```csharp
+var evt = new CloudEvent
+{
+    Id = "...",
+    ["xyz"] = "test"
```

Comment:
What's interesting here is that if the SDK supported:
```
["Id"] = "..."
```
too (meaning also allow all attributes to be defined using the same generic syntax and the SDK would check to see if the "xxx" is a known name or not) then this problem isn't as bad any more.